lib/oli/delivery/attempts/page_lifecycle/ungraded.ex

```diff
@@ -49,8 +55,50 @@ defmodule Oli.Delivery.Attempts.PageLifecycle.Ungraded do
   end
 
   @impl Lifecycle
-  def finalize(%FinalizationContext{} = _context) do
-    {:error, {:unsupported}}
+  def finalize(%FinalizationContext{
+        resource_attempt: %ResourceAttempt{lifecycle_state: :active} = resource_attempt,
+        datashop_session_id: datashop_session_id
+      }) do
+    # Collect all of the part attempt guids for all of the activities that get finalized
+    with {:ok, part_attempt_guids} <-
```

Comment:
This finalization can be much simpler: all that it needs to do is mark the attempt as `:evaluated`.  It should not evaluate existing active activities.  That will end up polluting analytics.